params/config.go

```diff
@@ -368,6 +371,70 @@ func (c *ChainConfig) AstriaExtraData() []byte {
 	return extra
 }
 
+type AstriaEIP1559Param struct {
+	MinBaseFee               uint64 `json:"minBaseFee"`
+	ElasticityMultiplier     uint64 `json:"elasticityMultiplier"`
+	BaseFeeChangeDenominator uint64 `json:"baseFeeChangeDenominator"`
+}
+
+type AstriaEIP1559Params struct {
+	heights        map[uint64]AstriaEIP1559Param
+	orderedHeights []uint64
+}
+
+func NewAstriaEIP1559Params(heights map[uint64]AstriaEIP1559Param) *AstriaEIP1559Params {
+	orderedHeights := []uint64{}
+	for k := range heights {
+		orderedHeights = append(orderedHeights, k)
+	}
+	sort.Slice(orderedHeights, func(i, j int) bool { return orderedHeights[i] > orderedHeights[j] })
+
+	return &AstriaEIP1559Params{
+		heights:        heights,
+		orderedHeights: orderedHeights,
+	}
+}
+
+func (c *AstriaEIP1559Params) MinBaseFeeAt(height uint64) *big.Int {
+	for _, h := range c.orderedHeights {
+		if height >= h {
+			return big.NewInt(0).SetUint64(c.heights[h].MinBaseFee)
+		}
+	}
+	return common.Big0
+}
+
+func (c *AstriaEIP1559Params) ElasticityMultiplierAt(height uint64) uint64 {
+	for _, h := range c.orderedHeights {
+		if height >= h {
+			return c.heights[h].ElasticityMultiplier
+		}
+	}
+	return DefaultElasticityMultiplier
+}
+
+func (c *AstriaEIP1559Params) BaseFeeChangeDenominatorAt(height uint64) uint64 {
+	for _, h := range c.orderedHeights {
+		if height >= h {
+			return c.heights[h].BaseFeeChangeDenominator
+		}
+	}
+	return DefaultBaseFeeChangeDenominator
+}
+
+func (c AstriaEIP1559Params) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.heights)
+}
+
+func (c *AstriaEIP1559Params) UnmarshalJSON(data []byte) error {
+	var heights map[uint64]AstriaEIP1559Param
+	if err := json.Unmarshal(data, &heights); err != nil {
+		return err
+	}
+	*c = *NewAstriaEIP1559Params(heights)
```

Comment:
does this actually work? or do you have to set each field within `c` manually? can you test this

Reply:
ya it works. I made a test script and it worked. I can add a few tests tho, prob not a bad idea.